scripting/get5/recording.sp

```diff
@@ -14,18 +14,45 @@ bool StartRecording() {
   }
 
   char demoName[PLATFORM_MAX_PATH + 1];
+
   if (!FormatCvarString(g_DemoNameFormatCvar, demoName, sizeof(demoName))) {
     LogError("Failed to format demo filename. Please check your demo file format convar.");
     g_DemoFileName = "";
     return false;
   }
 
-  Format(g_DemoFileName, sizeof(g_DemoFileName), "%s.dem", demoName);
+  char demoFolder[PLATFORM_MAX_PATH + 1];
+  char dateFormat[64];
+  char formattedDate[64];
+  g_DemoPathCvar.GetString(demoFolder, sizeof(demoFolder));
+  g_DateFormatCvar.GetString(dateFormat, sizeof(dateFormat));
+  int timeStamp = GetTime();
+  FormatTime(formattedDate, sizeof(formattedDate), dateFormat, timeStamp);
+
+  ReplaceString(demoFolder, sizeof(demoFolder), "{MATCHID}", g_MatchID);
+  ReplaceString(demoFolder, sizeof(demoFolder), "{DATE}", formattedDate, false);
+
+  int demoFolderLength = strlen(demoFolder);
+
+  if (demoFolderLength > 0 &&
```

Comment:
Turn this into a utility function as it's a duplicate of the logic used for backups.

Reply:
Moved and created new util function.